ports/stm32f4/boards/stm32f4_discovery/mpconfigboard.mk

```diff
@@ -0,0 +1,18 @@
+USB_VID = 0x239A
+USB_PID = 0x808A
+USB_PRODUCT = "STM32F407VG Discovery Board - CPy"
+USB_MANUFACTURER = "STMicroelectronics"
+USB_DEVICES = "CDC,MSC"
+
+INTERNAL_FLASH_FILESYSTEM = 1
+LONGINT_IMPL = NONE
+
+# This is technically a F407 but there's no difference
```

Comment:
Why not set to 407 below? This could be confusing if one searches for MCU_SUB_VARIANT and ignores a comment like this that is nearby.

Reply:
The MCU subvariant is used to associate with the package, the peripherals, and the startup file name - changing it to the f407 would mean having duplicate files and an extra check for those instances without actually being used for any changes.

Reply:
I'd recommend putting the proper 407 files in now even though we're not using the differences yet. Doing it now will make this file clearer until then and make it easier when someone wants to add support for the camera or ethernet. (camera may not be that far out if we want to support the openmv devices)

Reply:
Hmmm. This seems fine for now, but in the future it might be reasonable to group both of these under a generic "F40xxx" signifier like ST does, since they're so similar - the two MCUs actually share a datasheet, and the F407 is essentially just a footnote to include a couple extra peripherals on the F405.

Reply:
Though I guess then they'd be easy to mix up with the F401, which I guess ST just kind of ignored with that naming scheme?